src/components/markets/MarketFilter.tsx

```diff
@@ -0,0 +1,316 @@
+import {
+  Box,
+  IconButton,
+  InputAdornment,
+  Popover,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from '@mui/material';
+import React, { useState } from 'react';
+import { useTokenMetadataList } from '../../hooks/api';
+import { ToggleButton } from '../common/ToggleButton';
+import { TokenHeader } from '../common/TokenHeader';
+
+export type SortType = 'poolTvl' | 'backstopTvl';
+export type SortDirection = 'asc' | 'desc';
+
+export interface MarketFilterProps {
+  onFilterChange: (key: keyof MarketFilters, value: any) => void;
+  availableAssets: string[]; // List of available asset contract ids
+  defaultFilters: MarketFilters;
+}
+
+export interface MarketFilters {
+  search: string;
+  sortBy: SortType;
+  sortDirection: SortDirection;
+  selectedAssets: string[];
+}
+
+const sortByOptions: { value: SortType; label: string; direction: SortDirection }[] = [
+  { value: 'poolTvl', label: 'Pool TVL', direction: 'desc' },
+  { value: 'poolTvl', label: 'Pool TVL', direction: 'asc' },
+  { value: 'backstopTvl', label: 'Backstop TVL', direction: 'desc' },
+  { value: 'backstopTvl', label: 'Backstop TVL', direction: 'asc' },
+];
+
+export const MarketFilter: React.FC<MarketFilterProps> = ({
+  onFilterChange,
+  availableAssets,
+  defaultFilters,
+}) => {
+  const theme = useTheme();
+
+  const tokenMetadataList = useTokenMetadataList(availableAssets);
+
+  // Dropdown menu state
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+
+  // Handle changes by calling the parent's handler
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onFilterChange('search', event.target.value);
+  };
+
+  const handleSortByChange = (option: SortType, direction: SortDirection) => {
+    onFilterChange('sortBy', option);
+    onFilterChange('sortDirection', direction);
+  };
+
+  const handleAssetChange = (assetSymbol: string) => {
+    const newSelectedAssets = defaultFilters.selectedAssets.includes(assetSymbol)
+      ? defaultFilters.selectedAssets.filter((asset) => asset !== assetSymbol)
+      : [...defaultFilters.selectedAssets, assetSymbol];
+    onFilterChange('selectedAssets', newSelectedAssets);
+  };
+
+  const handleClearSearch = () => {
+    onFilterChange('search', '');
+  };
+
+  // Open filter dropdown
+  const openFilterMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  // Close filter dropdown
+  const closeFilterMenu = () => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <Box
+      sx={{
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+      }}
+    >
+      <Box
+        sx={{
+          flexGrow: 1,
+          flexShrink: 1,
+          minWidth: '100px',
+          maxWidth: '230px',
+          marginRight: 1,
+        }}
+      >
+        <TextField
```

Comment:
the disjointed edges on the circular part of the text box and the active underline looks a bit odd to me.

can the outline of the textbox have a border effect instead of just an underline?

for an example, the text boxes in github do this